File: 88/python3/main.py
#!/usr/bin/env python3

class Solution:
    def merge(self, nums1: list[int], m: int, nums2: list[int], n: int) -> None:
        # Truncate according to the expected elements
        nums1[:] = nums1[:m]
        nums2[:] = nums2[:n]

        # Emptiness checks
        if ((len(nums1) == 0 and len(nums2) == 0) or (m == 0 and n == 0)):
            nums1[:] = []
            return
        elif (m == 0 and n > 0):
            nums1[:] = nums2
            return
        elif (n == 0 and m > 0):
            return

        # Merge the two vectors into one
        n1_index = 0
        n2_index = 0
        combined_nums = list()
        while n1_index < m or n2_index < n:
            if n1_index >= m:
                combined_nums.append(nums2[n2_index])
                n2_index += 1
            elif n2_index >= n:
                combined_nums.append(nums1[n1_index])
                n1_index += 1
            elif nums1[n1_index] <= nums2[n2_index]:
                combined_nums.append(nums1[n1_index])
                n1_index += 1
            else:
                combined_nums.append(nums2[n2_index])
                n2_index += 1

        nums1[:] = combined_nums


# Test case 1
nums1 = [1, 2, 3, 0, 0, 0]
m = 3
nums2 = [2, 5, 6]
n = 3
Solution().merge(nums1, m, nums2, n)
print(nums1)  # Expected Output: [1, 2, 2, 3, 5, 6]

# Test case 2
nums1 = [1]
m = 1
nums2 = []
n = 0
Solution().merge(nums1, m, nums2, n)
print(nums1)  # Expected Output: [1]

# Test case 3
nums1 = [0]
m = 0
nums2 = [1]
n = 1
Solution().merge(nums1, m, nums2, n)
print(nums1)  # Expected Output: [1]

File: 88/cpp/main.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    void merge(std::vector<int>& nums1, int m, std::vector<int>& nums2, int n) {
        // Emptiness checks
        if ((nums1.size() == 0 && nums2.size() == 0) || (m == 0 && n == 0)) {
            nums1 = {};
            return;
        } else if (m == 0 && n > 0) {
            nums1 = nums2;
            return;
        } else if (n == 0 && m > 0) {
            return;
        }

        nums1.resize(m);
        nums2.resize(n);

        // Merge the two vectors into one
        int n1_index = 0;
        int n2_index = 0;
        std::vector<int> combined_nums;
        while (n1_index < m || n2_index < n) {
            if (n1_index >= m) {
                combined_nums.push_back(nums2[n2_index++]);
            } else if (n2_index >= n) {
                combined_nums.push_back(nums1[n1_index++]);
            } else if (nums1[n1_index] <= nums2[n2_index]) {
                combined_nums.push_back(nums1[n1_index++]);
            } else {
                combined_nums.push_back(nums2[n2_index++]);
            }
        }

        nums1 = combined_nums;
    }
};

int main() {
    Solution solution;
    
    // Example 1
    std::vector<int> nums1_1 = {1, 2, 3, 0, 0, 0};
    std::vector<int> nums2_1 = {2, 5, 6};
    solution.merge(nums1_1, 3, nums2_1, 3);
    for (int num : nums1_1) {
        std::cout << num << " ";
    }
    std::cout << std::endl;
    
    // Example 2
    std::vector<int> nums1_2 = {1};
    std::vector<int> nums2_2 = {};
    solution.merge(nums1_2, 1, nums2_2, 0);
    for (int num : nums1_2) {
        std::cout << num << " ";
    }
    std::cout << std::endl;

    // Example 3
    std::vector<int> nums1_3 = {0};
    std::vector<int> nums2_3 = {1};
    solution.merge(nums1_3, 0, nums2_3, 1);
    for (int num : nums1_3) {
        std::cout << num << " ";
    }
    std::cout << std::endl;

    return 0;
}
